src/test/java/it/gov/pagopa/reward/connector/event/consumer/HpanInitiaveConsumerConfigTest.java

```diff
@@ -198,15 +217,32 @@ private void checkValidMessages(int dbElementsNumbers, int updatedHpanNumbers) {
     void initializeDB(int requestElements){
         int initiativesWithCloseIntervals = requestElements/2;
         IntStream.range(0, initiativesWithCloseIntervals).
-                mapToObj(HpanInitiativesFaker::mockInstanceWithCloseIntervals)
+                mapToObj(i -> HpanInitiativesFaker.mockInstanceWithCloseIntervals(i, INITIATIVE_ID_PF))
                 .forEach(h -> hpanInitiativesRepository.save(h).subscribe(hSaved -> log.debug("saved hpan: {}", hSaved.getHpan())));
         IntStream.range(initiativesWithCloseIntervals, requestElements)
-                .mapToObj(HpanInitiativesFaker::mockInstance)
+                .mapToObj(i ->HpanInitiativesFaker.mockInstance(i, INITIATIVE_ID_PF))
                 .forEach(h -> hpanInitiativesRepository.save(h).subscribe(hSaved -> log.debug("saved hpan: {}", hSaved.getHpan())));
         initializeConcurrencyCase();
         initializeChannelManagementChannel(requestElements);
         waitForDB(requestElements+1+requestElements);
 
+        initializeFamily();
+
+    }
+
+    private void initializeFamily() {
+        List<OnboardingFamilies> ofList = IntStream.range(minBiasForNFInitiative+1, updatedHpanNumbers)
+                .mapToObj(i -> OnboardingFamiliesFaker.mockInstanceBuilder(i)
+                        .initiativeId(INITIATIVE_ID_NF)
+                        .memberIds(Set.of("USERID_%d".formatted(i))).build())
+                . toList();
+
+        ofList.forEach(of -> onboardingFamiliesRepository.save(of).subscribe(hSaved -> log.debug("saved family: {}", of.getFamilyId())));
+
+        long[] countSaved={0};
+        //noinspection ConstantConditions
+        TestUtils.waitFor(()->(countSaved[0]=hpanInitiativesRepository.count().block()) >= ofList.size(), ()->"Expected %d saved initiatives, read %d".formatted(ofList.size(), countSaved[0]), 60, 1000);
```

Comment:
why subscribe and then wait?

just block!
```suggestion

        onboardingFamiliesRepository.saveAll(ofList).collectList().block();
```